### PR TITLE
Add Idris Icons

### PR DESCRIPTION
**Pull request by zjhmale**
## Changes proposed:

- [x] Add
- [ ] Prepare
- [ ] Delete
- [ ] Fix

image

### CODE REVIEW

File: icons/file_type_idris.svg [hunk 1]
```diff
@@ -1 +1,62 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><title>file_type_idris</title><path d="M13.376,6.2c3.6,1.1,4.5,2,5.8,5.2-.3-4-1.9-5.6-5.8-5.2Z" style="fill:#8a0819"/><path d="M8.376,12.4c2.3.7,5,1.3,6,5.7.4-5.5-2.2-5.8-6-5.7Z" style="fill:#8a0819"/><path d="M9.976,8.8c3.5.7,5.7,1.7,7,6.2.3-5.8-2.9-6.4-7-6.2Z" style="fill:#8a0819"/><path d="M14.576,2c19.1,12.4-4.2,12.7.8,27.1l3,.9c-8.3-11.2,18.4-17-3.8-28Z" style="fill:#8a0819"/></svg>
\ No newline at end of file
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (JimiC) [inline icons/file_type_idris.svg]: This icon does not comply with our standards.

File: icons/file_type_idris2.svg [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (JimiC) [inline icons/file_type_idris2.svg]: This icon does not comply with our standards.

File: icons/file_type_ipkg.svg [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (JimiC) [inline icons/file_type_ipkg.svg]: This icon does not comply with our standards.

File: icons/file_type_ipkg2.svg [hunk 4]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (JimiC) [inline icons/file_type_ipkg2.svg]: This icon does not comply with our standards.

File: icons/file_type_idrisb.svg [hunk 5]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (JimiC) [inline icons/file_type_idrisb.svg]: This icon does not comply with our standards.

File: icons/file_type_idrisb2.svg [hunk 6]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (JimiC) [inline icons/file_type_idrisb2.svg]: This icon does not comply with our standards.

File: src/icon-manifest/supportedExtensions.ts [hunk 7]
```diff
@@ -166,7 +166,9 @@ export const extensions: IFileCollection = {
     { icon: 'haxecheckstyle', extensions: ['checkstyle.json'], filename: true, format: FileFormat.svg },
     { icon: 'haxedevelop', extensions: ['hxproj'], format: FileFormat.svg },
     { icon: 'html', extensions: [], languages: [languages.html], format: FileFormat.svg },
-    { icon: 'idris', extensions: ['idr', 'lidr', 'ibc'], format: FileFormat.svg },
+    { icon: 'idris', extensions: ['idr', 'lidr'], format: FileFormat.svg },
+    { icon: 'idrisb', extensions: ['ibc'], format: FileFormat.svg },
+    { icon: 'ipkg', extensions: ['ipkg'], format: FileFormat.svg },
```
Comment (JimiC) [inline src/icon-manifest/supportedExtensions.ts]: `icon` name should be `idrispkg`.

File: src/icon-manifest/supportedExtensions.ts [hunk 8]
```diff
@@ -166,7 +166,9 @@ export const extensions: IFileCollection = {
     { icon: 'haxecheckstyle', extensions: ['checkstyle.json'], filename: true, format: FileFormat.svg },
     { icon: 'haxedevelop', extensions: ['hxproj'], format: FileFormat.svg },
     { icon: 'html', extensions: [], languages: [languages.html], format: FileFormat.svg },
-    { icon: 'idris', extensions: ['idr', 'lidr', 'ibc'], format: FileFormat.svg },
+    { icon: 'idris', extensions: ['idr', 'lidr'], format: FileFormat.svg },
+    { icon: 'idrisb', extensions: ['ibc'], format: FileFormat.svg },
```
Comment (JimiC) [inline src/icon-manifest/supportedExtensions.ts]: `icon` name should not be `idrisb`. What is an `ibc` file?

**Comment by JimiC** [comment]
@zjhmale We prefer that the icons are generated via Illustrator. We realize that this isn't possible in many cases so as an alternative we prefer `Inkscape` `Optimized SVG` versions.

File: src/icon-manifest/supportedExtensions.ts (hunk 8 above)
Comment (zjhmale) [inline src/icon-manifest/supportedExtensions.ts]: `ibc` is compiled binary file here, I will change the icon name.

File: src/icon-manifest/supportedExtensions.ts (hunk 7 above)
Comment (zjhmale) [inline src/icon-manifest/supportedExtensions.ts]: OK.

**Comment by zjhmale** [comment]
Hi @JimiC, I generate those svg files using `Inkscape` and save them as `Inkscape Optimized SVG`, so how can I remove those image tags from `Inkscape`? Is there any simple steps to indicate how to generate a standard `vscode-icons` icon using `Inkscape`? Thx in advance 😄

**Comment by zjhmale** [comment]
I will pick up an standard icon and modify the XML source directly.

File: src/icon-manifest/supportedExtensions.ts (hunk 8 above)
Comment (JimiC) [inline src/icon-manifest/supportedExtensions.ts]: In that case, it should be named `idrisbin`.

**Comment by JimiC** [comment]
This is why I don't like working with `Inkscape`. There are no simple steps.
The basic standards for the icons we use are:
 - 32x32 px
 - 2px margin

File: icons/file_type_ipkg2.svg (hunk 4 above)
Comment (robertohuertasm) [inline icons/file_type_ipkg2.svg]: Why do you have the 2 version of all the icons?

File: icons/file_type_idris.svg (hunk 1 above)
Comment (robertohuertasm) [inline icons/file_type_idris.svg]: The original icon was using only paths. The new ones is using an image. The path should be reused.

Comment (robertohuertasm) [inline icons/file_type_idris.svg]: I'm pushing the icons reworked with Illustrator to your repo.

**Comment by zjhmale** [comment]
@robertohuertasm Thanks for the kind help 😄 I'm in UTC+8 and just planned to improve the PR today. Thanks again!

**Pull request merged**